snapshots/overlay/overlay.go

```diff
@@ -464,6 +481,10 @@ func (o *snapshotter) mounts(s storage.Snapshot) []mount.Mount {
 		options = append(options, "userxattr")
 	}
 
+	if o.volatile {
+		options = append(options, "volatile")
```

Comment:
overlayfs snapshot is service and this option is global setting. I think there are existing cases which required the rootfs is synced after umount. Currently, the kubernetes container is temporary and we can't restart a exited container. kubernetes always recreates container for pod. So, I would like to add the `volatile` option from caller, for example, CRI plugin.

I was thinking that we can go in this way:

a. When the overlayfs snapshot inits, it use `SupportsVolatileMount` to check and add the capability in Meta.

https://github.com/containerd/containerd/blob/e52fbfd5c16b74826f02bfdd96f99e22da7c2d24/snapshots/overlay/plugin/plugin.go#L41-L61

b. Add the mount option to add `volatile` if snapshot has the capability. Currently, the CRI UserNS feature is working with capability. You can check the main branch for this. And just in case, we should add CRI option to turn this feature on.